sqldelight-compiler/src/main/kotlin/app/cash/sqldelight/core/compiler/DatabaseGenerator.kt

```diff
@@ -72,7 +78,10 @@ internal class DatabaseGenerator(
 
     // Database constructor parameter:
     // driver: SqlDriver
-    val dbParameter = ParameterSpec.builder(DRIVER_NAME, dialect.driverType).build()
+    if (generateAsync) {
+      check(dialect.asyncDriverType != null) { "Dialect $dialect does not support async drivers" }
```

Comment:
when you've promoted the default type just search for these checks and remove them all